compiler-core/src/type_/tests/warnings.rs

```diff
@@ -434,6 +434,38 @@ fn unused_imported_module_with_alias_warnings_test() {
     );
 }
 
+// https://github.com/gleam-lang/gleam/issues/2326
+#[test]
+fn unused_imported_module_with_alias_and_unqualified_name_warnings_test() {
+    let warnings = get_warnings(
+        "import gleam/foo.{bar} as foo",
+        vec![("thepackage", "gleam/foo", "pub fn bar() { 1 }")],
+    );
+    assert!(!warnings.is_empty());
+    assert_eq!(
+        Warning::UnusedImportedValue {
```

Comment:
Perhaps this warning should be omitted in that situation?

Reply:
Oh maybe! If we can cleanly do that.

Reply:
What I had trouble with in `handle_unused`, is that the imported names are no longer associated with their import, so it's difficult to check if a whole import statement is unused. I think we would need to add the original module name in the EntityKind as proposed here: https://github.com/gleam-lang/gleam/blob/8d8af276334f78c92a7011897702809a18aabf91/compiler-core/src/type_/environment.rs#L94-L98

What do you think about that?

Reply:
I don't think we should make this any more complex as it will be replace with a proper call graph traversal in future.

Reply:
Alright, so until then, is it ok to emit the extra warning as shown in the test? If so then I think this PR is ready, and I would rebase #2321 on top of this.

Reply:
Yup sounds good 👍